<?xml version="1.0"?>
<launch>
    <!--<arg name="dev" default="/dev/ttyACM4"/>-->
    <arg name="baud" default="115200"/>
    <!--<arg name="address" default="128"/>-->
    <arg name="rate" default="20"/>
    <arg name="max_speed" default="2.0"/>
    <arg name="ticks_per_rotation" default="2000"/>      <!-- 66390.8 -->
    <arg name="gear_ratio" default="45.5"/>
    <arg name="wheel_circ" default="0.964"/>
    <arg name="base_width" default="0.69"/>
    <arg name="run_plow" default="true"/>
    <arg name="publish_odom" default="true"/>
    <arg name="publish_tf" default="false"/>
    <arg name="publish_currents" default="false"/>
    <arg name="stop_movement" default="true"/>
    <arg name="max_accel" default="1.5"/> <!--accel est en ms^2-->

    <node pkg="roboclaw_node" type="roboclaw_motors_single.py" name="roboclaw_motors" output="screen" respawn="true">
        <param name="~dev" value="/dev/rcMotors"/>
        <param name="~baud" value="$(arg baud)"/>
        <param name="~address" value="128"/>
        <param name="~rate" value="$(arg rate)"/>
        <param name="~max_speed" value="$(arg max_speed)"/>
        <param name="~ticks_per_rotation" value="$(arg ticks_per_rotation)"/>
        <param name="~gear_ratio" value="$(arg gear_ratio)"/>
        <param name="~wheel_circ" value="$(arg wheel_circ)"/>
        <param name="~base_width" value="$(arg base_width)"/>
        <param name="~publish_odom" value="$(arg pub_odom)"/>
        <param name="~publish_tf" value="$(arg publish_tf)"/>
        <param name="~publish_currents" value="$(arg pub_currents)"/>
        <param name="~stop_movement" value="$(arg stop_movement)"/>
        <param name="~max_accel" value="$(arg max_accel)"/>
        <param name="~name" value="roboclaw_motor"/>
        <remap from="cmd_vel" to="/$(arg robot)/control/cmd_vel"/>
    </node>

    <node if="$(arg run_plow)" pkg="roboclaw_node" type="roboclaw_plow.py" name="roboclaw_plow" output="screen" respawn="true">
        <param name="~dev" value="/dev/rcPlow"/>
        <param name="~baud" value="$(arg baud)"/>
        <param name="~address" value="130"/>
        <param name="~rate" value="$(arg rate)"/>
        <param name="~max_duty" value="100"/>
        <param name="~name" value="roboclaw_plow"/>
        <remap from="cmd_vel_plow" to="/$(arg robot)/control/cmd_vel_plow"/>
    </node>

    <!-- <node pkg="diagnostic_aggregator" type="aggregator_node"
        name="diagnostic_aggregator">
        <rosparam command="load"
                file="$(find roboclaw_node)/config/roboclaw_diag.yaml"/>
    </node> -->


</launch>
